### ISSUE TITLE
eachLike without object name only produces one child

### ISSUE THREAD

**Issue by bengro** [DiUS/pact-jvm#555]
I would like to model my mock such that its body consists of an unnamed array of n elements. The below code using `eachLike` only creates one element in the array, when it should produce three.

```
withBody(eachLike(3, {
   id identifier
   state("COMPLETED")
   type regexp("(A|B)", "A")
 }))
```

Oddly enough, if I name the name collection, it actually generates three objects:
```
withBody {
  namedObject eachLike(3, {
  id identifier
  })
}
```

This produces an array named `namedObject` with three children.

Where do I go wrong? Is this a bug?

**Comment by pturovsk** [comment]
Any update on this one? Getting the same issue

_Dependency version_: `"au.com.dius:pact-jvm-consumer-groovy_2.12:3.6.3"`

**Comment by uglyog** [comment]
This fix has been released with version 3.6.4
